### PR TITLE
Refactored K8s Pod node selector

### PR DESCRIPTION
**Pull request by zc-devs**
- using existing map instead of an array
- applying var name convention
- using arch constant

### CODE REVIEW

**Comment by zc-devs** [comment]
Pod config:
```
...
  containers:
    - name: wp-01h49cmzpzapdq3r1bkqbrttxa-0-step-0
      image: alpine
      ...
      imagePullPolicy: Always
      ...
  nodeSelector:
    kubernetes.io/arch: amd64
  ...

```

File: pipeline/backend/kubernetes/pod.go [hunk 1]
```diff
@@ -92,18 +93,19 @@ func Pod(namespace string, step *types.Step, labels, annotations map[string]stri
...
 	labels["step"] = podName
 
-	var platform string
-	for _, e := range mapToEnvVars(step.Environment) {
-		if e.Name == "CI_SYSTEM_ARCH" {
-			platform = e.Value
-			break
-		}
+	var nodeSelector map[string]string
+	if platform, exist := step.Environment["CI_SYSTEM_ARCH"]; exist {
```
Comment (6543) [inline pipeline/backend/kubernetes/pod.go:97]: step.platform ?

Comment (6543) [reply pipeline/backend/kubernetes/pod.go:97]: todo for me: check if this was not a bug in the first place :)

Comment (zc-devs) [reply pipeline/backend/kubernetes/pod.go:97]: > step.platform

There is no such field in step model.

Also there is deprecation warning in [environment.go](https://github.com/woodpecker-ci/woodpecker/blob/ae2bfe666f08d065ee4de0a22a8f56f747df9445/pipeline/frontend/metadata/environment.go#L116):
```
// DEPRECATED
"CI_SYSTEM_ARCH": m.Sys.Platform, // TODO: remove after v1.0.x version
```

**Review by 6543 — APPROVED** [review]

**Pull request merged**